src/test/kotlin/uk/gov/dluhc/printapi/rest/GenerateTemporaryCertificateIntegrationTest.kt

```diff
@@ -63,8 +81,76 @@ internal class GenerateTemporaryCertificateIntegrationTest : IntegrationTest() {
                 )
             )
             .contentType(MediaType.APPLICATION_JSON)
+            .body(
+                Mono.just(buildGenerateTemporaryCertificateRequest()),
+                GenerateTemporaryCertificateRequest::class.java
+            )
             .exchange()
             .expectStatus()
             .isForbidden
     }
+
+    @Test
+    fun `should return bad request given JSR303 validation failure`() {
+        // Given
+        wireMockService.stubCognitoJwtIssuerResponse()
+        val invalidGssCode = "invalid GSS Code"
+        val requestBody = buildGenerateTemporaryCertificateRequest(
+            gssCode = invalidGssCode
+        )
+
+        // When
+        val response = webTestClient.post()
+            .uri(URI_TEMPLATE, ERO_ID)
+            .bearerToken(getBearerToken(eroId = ERO_ID, groups = listOf("ero-$ERO_ID", "ero-vc-admin-$ERO_ID")))
+            .contentType(MediaType.APPLICATION_JSON)
+            .body(
+                Mono.just(requestBody),
+                GenerateTemporaryCertificateRequest::class.java
+            )
+            .exchange()
+            .expectStatus()
+            .isBadRequest
+            .returnResult(ErrorResponse::class.java)
+
+        // Then
+        val actual = response.responseBody.blockFirst()
+        assertThat(actual)
+            .hasStatus(400)
+            .hasError("Bad Request")
+            .hasMessageContaining("Validation failed for object='generateTemporaryCertificateRequest'. Error count: 1")
+            .hasValidationError("Error on field 'gssCode': rejected value [$invalidGssCode], size must be between 9 and 9")
+    }
+
+    @Test
+    fun `should return bad request given initBinder validation failure`() {
```

Comment:
Usually the method names are more Business focussed (this one and the JSR303 seem too technical)

Reply:
Yes, agreed. Now that I've removed the initBinder validation, and we only have the JSR303 validation, I've renamed that test so that it simply talks about an invalid request 👍